src/util.c

```diff
@@ -20,8 +20,14 @@ writable_buffer_ensure_space(writable_buffer *wb,
       minspace = 2 * wb->buffer_capacity;
    }
 
-   void *oldptr  = wb->can_free ? wb->buffer : NULL;
-   void *newdata = platform_realloc(wb->heap_id, oldspace, oldptr, minspace);
+   void *oldptr = wb->can_free ? wb->buffer : NULL;
+
+   // NOTE: realloc() may adjust-up 'minspace' for alignment
+   // ALSO: We must supply correct size of fragment being freed, which
+   //   will be the buffer capacity, but not just the oldspace in-use.
+   //   (Otherwise, free-fragment management will run into memory leaks.)
+   void *newdata =
+      platform_realloc(wb->heap_id, wb->buffer_capacity, oldptr, &minspace);
```

Comment:
Important bug-fix that call to `realloc` should pass buffer's capacity (i.e., old memory fragment's size) but not `oldspace` (which could be far less than capacity).